File: benchmarks/java/integer_boolean/Clojure_Big_Integer/Multiply.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class Multiply {






    private BInteger counter;
    private BInteger value;

    public Multiply() {
        counter = (BInteger) new BInteger("0");
        value = (BInteger) new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("5000000"))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger("1"));
            value = (BInteger) value.multiply(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        Multiply exec = new Multiply();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: benchmarks/cpp/integer_boolean/primitive_integer/Multiply.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef Multiply_H
#define Multiply_H

using namespace std;

class Multiply {



    private:



        BInteger counter;
        BInteger value;

    public:

        Multiply() {
            counter = (BInteger(0));
            value = (BInteger(0));
        }

        void simulate() {
            while((counter.less((BInteger(5000000)))).booleanValue()) {
                counter = counter.plus((BInteger(1)));
                value = value.multiply((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    Multiply exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

